actors/miner/tests/util.rs

```diff
@@ -204,8 +261,1024 @@ impl ActorHarness {
+            sectors: all_sector_numbers,
+            reward_smoothed: self.epoch_reward_smooth.clone(),
+            reward_baseline_power: self.baseline_power.clone(),
+            quality_adj_power_smoothed: self.epoch_qa_power_smooth.clone(),
+        };
+        rt.call::<Actor>(
+            Method::ConfirmSectorProofsValid as u64,
+            &RawBytes::serialize(params).unwrap(),
+        )
+        .unwrap();
+        rt.verify();
+    }
+
+    fn confirm_sector_proofs_valid_internal(
+        &self,
+        rt: &mut MockRuntime,
+        cfg: ProveCommitConfig,
+        pcs: &Vec<SectorPreCommitOnChainInfo>,
+    ) {
+        let mut valid_pcs = Vec::new();
+        for pc in pcs {
+            if pc.info.deal_ids.len() > 0 {
+                let params = ActivateDealsParams {
+                    deal_ids: pc.info.deal_ids.clone(),
+                    sector_expiry: pc.info.expiration,
+                };
+
+                let mut exit = ExitCode::Ok;
+                match cfg.verify_deals_exit.get(&pc.info.sector_number) {
+                    Some(exit_code) => {
+                        exit = exit_code.clone();
+                    }
+                    None => {
+                        valid_pcs.push(pc);
+                    }
+                }
+
+                rt.expect_send(
+                    *STORAGE_MARKET_ACTOR_ADDR,
+                    MarketMethod::ActivateDeals as u64,
+                    RawBytes::serialize(params).unwrap(),
+                    TokenAmount::from(0),
+                    RawBytes::default(),
+                    exit,
+                );
+            } else {
+                valid_pcs.push(pc);
+            }
+        }
+
+        if valid_pcs.len() > 0 {
+            let mut expected_pledge = TokenAmount::from(0);
+            let mut expected_qa_power = BigInt::from(0);
+            let mut expected_raw_power = BigInt::from(0);
+
+            for pc in valid_pcs {
+                let pc_on_chain = self.get_precommit(rt, pc.info.sector_number);
+                let duration = pc.info.expiration - rt.epoch;
+                if duration >= rt.policy.min_sector_expiration {
+                    let qa_power_delta = qa_power_for_weight(
+                        self.sector_size,
+                        duration,
+                        &pc_on_chain.deal_weight,
+                        &pc_on_chain.verified_deal_weight,
+                    );
+                    expected_qa_power += &qa_power_delta;
+                    expected_raw_power += self.sector_size as u64;
+                    let mut pledge = initial_pledge_for_power(
+                        &qa_power_delta,
+                        &self.baseline_power,
+                        &self.epoch_reward_smooth,
+                        &self.epoch_qa_power_smooth,
+                        &rt.total_fil_circ_supply(),
+                    );
+
+                    if pc_on_chain.info.replace_capacity {
```

Comment:
Since this behavior is now strictly deprecated following snap deals I advocate for removing this pledge calculation and instead failing the test.  We shouldn't have anything that makes it this far.  The furthest we should now get is precommit where this should fail.  Tests of the old behavior should not be duplicated from go actors.

Reply:
ah ok, i didnt know. Yeah, lets fix this.

Reply:
i'll also leave a comment for the deviation.

Reply:
I replaced the pledge calculation with an assertion:
```
assert!(replaced.initial_pledge <= pledge);
```